### PR TITLE
 docs(tutorials): complete the storage step

### LINKED ISSUE #94: docs(tutorials): complete the storage step
complete [step 6 Set-up Storage](https://next.blindnet.dev/docs/tutorials/intro/storage-set-up) of the introductory tutorial

### PR DESCRIPTION
**Pull request by nweldev**
close https://github.com/blindnet-io/devrel-management/issues/94

BLOCKED by https://github.com/blindnet-io/product-management/issues/983

### CODE REVIEW

**Comment by noelmace** [comment]
@TheKinrar: this step is now ready for review, but I'll keep it as a draft until we add the missing parameter in the PCE.

Just ignore the related warning.

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 1]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+1. a [Data Access Component](/docs/storage/data-access) (DAC)
+2. a storage service, with optionnal help from a devkit storage connector library (currently, only Java is supported, via the  [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java))
 
-You can choose to use the [Capture Component](/docs/data-capture) and directly store the captured data in blindnet's preconfigured storage, or your own storage service.
+You can choose to use the [Capture Component](/docs/data-capture) and directly store the captured data in blindnet's preconfigured storage, or your own storage service. You can also directly connect the Data Access Component to your own storage service without using the Capture Component. For see sake of simplicity, we'll choose this option for this tutorial.
 
-You can also directly connect the Data Access Component to your own storage service without using the Capture Component. For see sake of simplicity, we'll choose this option for this tutorial.
+For that, we'll set up a dedicated demo storage service in Java, and connect it to the Capture Component using the.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:14]:
```suggestion
For that, we'll set up a dedicated demo storage service in Java, and connect it to the Data Access Component using the [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java).
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 2]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
 
-> :construction_worker: writing in progress
+The blindnet devkit storage can be handled via two components:
 
-A [Data Access Component](/docs/storage/data-access) is here to transform requests from the [Privacy Computation Engine](/docs/computation) to efficiently execute automated operations on your data storage.
+1. a [Data Access Component](/docs/storage/data-access) (DAC)
+2. a storage service, with optionnal help from a devkit storage connector library (currently, only Java is supported, via the  [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java))
 
-You can choose to use the [Capture Component](/docs/data-capture) and directly store the captured data in blindnet's preconfigured storage, or your own storage service.
+You can choose to use the [Capture Component](/docs/data-capture) and directly store the captured data in blindnet's preconfigured storage, or your own storage service. You can also directly connect the Data Access Component to your own storage service without using the Capture Component. For see sake of simplicity, we'll choose this option for this tutorial.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:12]:
```suggestion
You can choose to use the [Capture Component](/docs/data-capture) and directly store the captured data in blindnet's preconfigured storage, or your own storage service. You can also directly connect the Data Access Component to your own storage service without using the Capture Component. For the sake of simplicity, we'll choose this option for this tutorial.
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 3]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
 tags: [tutorial]
 ---
 
-> :construction_worker: writing in progress
+The blindnet devkit storage can be handled via two components:
 
-A [Data Access Component](/docs/storage/data-access) is here to transform requests from the [Privacy Computation Engine](/docs/computation) to efficiently execute automated operations on your data storage.
+1. a [Data Access Component](/docs/storage/data-access) (DAC)
+2. a storage service, with optionnal help from a devkit storage connector library (currently, only Java is supported, via the  [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java))
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:10]:
```suggestion
2. a storage service, with optional help from a devkit storage connector library (currently, only Java is supported, via the  [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java))
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 4]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
 
-You can also directly connect the Data Access Component to your own storage service without using the Capture Component. For see sake of simplicity, we'll choose this option for this tutorial.
+For that, we'll set up a dedicated demo storage service in Java, and connect it to the Capture Component using the.
 
-For that, we'll set up a dedicated demo storage service in Java, and connect it to the Capture Component using the [Java Storage Connector library](https://github.com/blindnet-io/storage-connector-java).
+## Pre-requisites
+
+The following steps require **git**, a **Java Development Kit**, **sbt**, and **docker** to be installed on your machine.
+
+Refer to the [associated section](http://localhost:3000/docs/tutorials/intro/PCE-set-up#pre-requisites) of the first step of this tutorial for more information.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:20]: I guess this localhost link should be fixed

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 5]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+Additionnally, [Maven](https://maven.apache.org/install.html) is also required in order to build the associated demo.
+
+
+## Running the Data Access Component Locally
+
+The [Data Access Component](/docs/storage/data-access) act as a proxy between the  [Privacy Computation Engine](/docs/computation) and your storage servicde, transforming requests from the PCE to efficiently execute automated operations on your data storage.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:27]:
```suggestion
The [Data Access Component](/docs/storage/data-access) acts as a proxy between the  [Privacy Computation Engine](/docs/computation) and your storage service, transforming requests from the PCE to efficiently execute automated operations on your data storage.
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 6]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+[io-compute-1] INFO org.http4s.blaze.server.BlazeServerBuilder - 
+  _   _   _        _ _
+ | |_| |_| |_ _ __| | | ___
+ | ' \  _|  _| '_ \_  _(_-<
+ |_||_\__|\__| .__/ |_|/__/
+             |_|
+[io-compute-1] INFO org.http4s.blaze.server.BlazeServerBuilder - http4s v0.23.14 on blaze v0.23.12 started at http://127.0.0.1:8028/
+```
+
+Your DAC instance is now ready to receive request from a PCE.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:62]:
```suggestion
Your DAC instance is now ready to receive requests from a PCE.
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 7]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+:::
+
+## Storage Connector Demo
+
+### Set Up
+
+For the purpose of this tutorial, we provide a demo implementation of a storage service.
+
+Code for this demo can be fund in the `example` directory in the librarie's repository: [github.com/blindnet-io/storage-connector-java > `example`](https://github.com/blindnet-io/storage-connector-java/tree/develop/example).
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:76]:
```suggestion
Code for this demo can be found in the `example` directory in the library repository: [github.com/blindnet-io/storage-connector-java > `example`](https://github.com/blindnet-io/storage-connector-java/tree/develop/example).
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 8]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+Clone this resporitoy:
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:78]:
```suggestion
Clone this repository:
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 9]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+The `StorageConnector` class is the center of the Java Storage Connector library, instanciated via the `create()` method.
+
+```java
+import io.blindnet.storageconnector.StorageConnector;
+
+StorageConnector sc = StorageConnector.create();
+```
+
+This class allows you to establish a transparent connection with the PCE to answer in the most efficient way to every request the PCE send.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:126]:
```suggestion
This class allows you to establish a transparent connection with the PCE to answer in the most efficient way every request the PCE sends.
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 10]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+This connection is only active after calling the `start()` method.
+
+```java
+StorageConnector.create().start();
+```
+
+Yet, the default implementation of `StorageConnector` doesn't make any assumption about your selectors or storage.
+Meaning the `StorageConnector` in the above example will only have a `DefaultDataRequestHandler`, which deny all incoming requests.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:134]:
```suggestion
Meaning the `StorageConnector` in the above example will only have a `DefaultDataRequestHandler`, which denies all incoming requests.
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 11]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+                    rs.getBytes("proof")
+            );
+        }
+    }
+}
+```
+
+:::info
+
+This example use `jsdbi`, but the Storage Connector library is compatible with any ORM solution, even project-specific ones.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:179]:
```suggestion
This example uses `jdbi`, but the Storage Connector library is compatible with any ORM or ORM-less storage solution, even project-specific ones.
```

Comment (TheKinrar) [reply docs/02-tutorials/01-intro/07-storage-set-up.mdx:179]: (jdbi isn't really an ORM)

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 12]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+MappingRequestHandler userRequestHandler = userMappingBuilder.build();
+
+storageConnector.setDataRequestHandler(userRequestHandler).start();
+```
+
+#### Mapping Simple Entries
+
+First, we start by associating our mapper with a specific stored data.
+For that, we use the `Builder.setSubjectMapper()` method, which only need a function returning the desired object based on a given selector (here, the user's email).
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:208]:
```suggestion
For that, we use the `Builder.setSubjectMapper()` method, which only needs a function returning the desired object based on a given selector (here, the user's email).
```

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx [hunk 13]
```diff
@@ -4,16 +4,239 @@ title: 6. Set-up Storage
...
+
+```java
+userMappingBuilder.addSelectorType("NAME", User::fullName)
+```
+
+##### Mapping Binary Data
+
+The Java Storage Connector library also provides all the methods you need to map binary data in order to handle them in the more efficient way.
+
+Here, the demonstration app also capture a proof of identity document (pdf, image, etc.) for every user, associating it with the custom `OTHER-DATA.PROOF` selector.
```
Comment (TheKinrar) [inline docs/02-tutorials/01-intro/07-storage-set-up.mdx:233]:
```suggestion
Here, the demonstration app also captures a proof of identity document (pdf, image, etc.) for every user, associating it with the custom `OTHER-DATA.PROOF` selector.
```

**Review by TheKinrar — CHANGES_REQUESTED** [review]
Stumbled upon some typos, thought I would also report them, the result is this review is mostly about typos.

We'll have to make some changes to the tutorial(s) in like a week or two when we start using authentication, app scopes and the identity server, but I'd say this is fine for now!

File: docs/02-tutorials/01-intro/07-storage-set-up.mdx (hunk 4 above)
Comment (noelmace) [reply docs/02-tutorials/01-intro/07-storage-set-up.mdx:20]: indeed

```suggestion
Refer to the [associated section](https://blindnet.dev/docs/tutorials/intro/PCE-set-up#pre-requisites) of the first step of this tutorial for more information.
```

**Comment by noelmace** [comment]
Thank you very much @TheKinrar. You are perfectly right. IMHO, the best option would be to split this tutorial into multiple component-specific ones and to replace it for real quick start either with a very simple one using the devkit monorepo once the docker-compose config is done to underline the on-premise approach, or a completely new SaaS approach.

**Comment by TheKinrar** [comment]
I agree. Full self-hosting means at least four components, so we clearly need a docker-compose for that. Besides, it makes things easier, for us, and for users.

**Comment by TheKinrar** [comment]
Superseded by #90.

**Pull request closed without merging**